gaps/gap-4_comp_manifest/gap-4_comp_manifest.md

```diff
@@ -107,9 +107,17 @@ from the ground up using properties only.
 
 7. `golem.srv.comp.manifest.net.inet.out.urls : List[String]`
 
-    List of allowed external URLs that outbound requests can be sent to.
+    List of allowed external URLs that outbound requests can be sent to.  
     E.g. `["http://golemfactory.s3.amazonaws.com/file1", "http://golemfactory.s3.amazonaws.com/file2"]`
 
+    If unrestricted outbound is requested this property must not be set.
+
+8. `golem.srv.comp.manifest.net.inet.out.unrestricted : true`
+
+   This property means that the payload requires unrestricted outbound access. When present the value is always `true`. Either this property or the URL list in `golem.srv.comp.manifest.net.inet.out.urls` must be present.
+
+   If neither `golem.srv.comp.manifest.net.inet.out.unrestricted` nor `golem.srv.comp.manifest.net.inet.out.urls` is present, the manifest should be considered invalid and outbound access should not be permitted.
```

Comment:
Should it be `golem.srv.comp.manifest.net.inet.out.unrestricted` or `golem.srv.comp.manifest.net.inet.out.urls.unrestricted`?

Reply:
I wanted to avoid having `golem.srv.comp.manifest.net.urls` to parse into either a list or other type. Your suggestion would mean that this property can be either a list or an object.
I chose `golem.srv.comp.manifest.net.inet.out.unrestricted` specifically to avoid ambiguity.

Reply:
I understand this and would like to avoid it as well.
But if you would have any other type of restrictions, then you can't mark them as `unrestricted` separately.

Reply:
So lets say we introduce bandwidth limits. Then we can add it in two ways:
```
"golem.srv.comp.manifest.net.inet.out.bandwidth" : {
  "oneOf": [
    { "type": "string", "const": "unrestricted" },
    { "type": "object", "description": "detailed limits", "properties": { } }
  ]
}
```
We are introducing a new property so it is not a problem that it can parse into a string or into an object. I wanted to avoid it in a case where it is already out.
The other way is to introduce a new property:
`golem.srv.comp.manifest.net.inet.out.unlimited-bandwidth`.

Maybe the wording is not good and instead of `unrestricted` it could be called `without-url-filtering` or something similar. WDYT?

Reply:
> I wanted to avoid it in a case where it is already out.

You are right. We should avoid it.
So maybe `unrestricted-urls`?

Reply:
How about:
`golem.srv.comp.manifest.net.inet.out.unrestricted.urls = true`
So this 'unrestricted' namespace can be easily extended for other limitations introduced later.

Reply:
I would prefer your proposal from this comment in future use cases: https://github.com/golemfactory/golem-architecture/pull/75#discussion_r1171549618
`golem.srv.comp.manifest.net.inet.out.unrestricted.urls = true` requires more validation.
Because someone can set it to `false` without providing urls list or set it to true and provide the list.
Your previous proposal doesn't have this problems, so for future use cases I would use it.

For this specific case I'm ok with using `golem.srv.comp.manifest.net.inet.out.unrestricted.urls = true`